lib/fluent/plugin/out_google_cloud.rb

```diff
@@ -1224,6 +1224,30 @@ def group_log_entries_by_tag_and_local_resource_id(chunk)
     # collection of entries.
     def determine_group_level_monitored_resource_and_labels(tag,
                                                             local_resource_id)
+      # Determine the monitored resource based on the local_resource_id.
+      # Different monitored resource types have unique ids in different format.
+      # We will query Metadata Agent for the monitored resource. Return the
+      # legacy monitored resource (either the instance resource or the resource
+      # inferred from the tag) if failed to get a monitored resource from
+      # Metadata Agent with this key.
+      #
+      # Examples:
+      # "container.<container_id>" // Docker container.
+      # "k8s_pod.<namespace_name>.<pod_name>" // GKE pod.
+      if local_resource_id
+        resource = monitored_resource_from_local_resource_id(local_resource_id)
+        if resource
+          common_labels = determine_common_labels_for_resource_type(
+            resource.type)
+
+          resource.freeze
+          resource.labels.freeze
+          common_labels.freeze
+
+          return [resource, common_labels]
```

Comment:
What happens to `gke_container` when we return early? Additional common labels like `namespace_name` and `pod_name` will be missing (https://github.com/GoogleCloudPlatform/fluent-plugin-google-cloud/pull/243/files#diff-0d6486c76b449a4c9242958738de9cdbL1296).

As of today, this is not a problem because we don't use Metadata Agent for legacy `gke_container` resource, but Metadata Agent does support it. Should we explicitly exclude it?

Reply:
This is only returned when the local resource is discovered via the metadata agent, the metadata agent should return the "correct" monitored resource.

Reply:
I misunderstood this, I've duplicated the logic for copying common labels as we discussed.